overlays/utils/default.nix

```
@@ -16,16 +16,13 @@ in {
     GC_DONT_GC=1 ${nixFlakes}/bin/nix --experimental-features "nix-command flakes" "$@"
   '';
   cabalWrapped = writeShellScriptBin "cabal" ''
-    >&2 echo 'Temporary modify `cabal.project` to use nix builds of `source-repository-package`s (haskell/cabal#5444 and haskell/cabal#6249).'
-    PROJECT="$(${git}/bin/git rev-parse --show-toplevel)/cabal.project"
-    BACKUP="$(mktemp)"
-    cp -a "$PROJECT" "$BACKUP"
-    sed -ni '1,/--- 8< ---/ p' $PROJECT
-    function atexit() {
-        mv "$BACKUP" "$PROJECT"
-    }
-    trap atexit EXIT
-    ${final.cabal or final.cabal-install}/bin/cabal "$@"
+    TOPLEVEL="$(${git}/bin/git rev-parse --show-toplevel)"
+    ORIG_PROJECT="$TOPLEVEL/cabal.project"
+    # Should be in .gitignore:
+    NIXSHELL_PROJECT="$TOPLEVEL/.nix-shell-cabal.project"
+    
+    sed -n '1,/--- 8< ---/ p' $PROJECT > "$NIXSHELL_PROJECT"
```

Should it be ORIG_PROJECT instead of PROJECT?

And can we please use a more clever filter? Like this:
```suggestion
    awk '
  # Matches all section starts
  /^[^ ]/ {
    # Remember the section name (they can span multiple lines)
    section = $0
  }
  # Matches every line
  // {
    # Only print the line if it is not in the section we want to omit
    if (section != "source-repository-package")
      print $0
  }
' "$ORIG_PROJECT" > "$NIXSHELL_PROJECT"
```